indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java

```diff
@@ -97,13 +102,22 @@ public boolean isReady(TaskActionClient taskActionClient)
   @Override
   public void stopGracefully(TaskConfig taskConfig)
   {
+    aborted.set(true);
   }
 
   @Override
   public TaskStatus runTask(TaskToolbox toolbox) throws Exception
   {
-    Thread.sleep(runTime);
-    return TaskStatus.success(getId());
+    toolbox.getEmitter().emit(ServiceMetricEvent.builder().setMetric(NOOP_TASK_EVENT_STARTED, 1));
+    long endTime = System.currentTimeMillis() + runTime;
+    while (endTime > System.currentTimeMillis() && !aborted.get()) {
```

Comment:
Just curious as to why we needed to maintain the `aborted` field.
When we stop the `Indexer`, it should ideally interrupt all its tasks, thus triggering a `task/run/time` `FAILED` event.
Was that not happening?